gtsam/3rdparty/CMakeLists.txt

```diff
@@ -81,7 +81,7 @@ if(GEOGRAPHICLIB_FOUND AND GTSAM_INSTALL_GEOGRAPHICLIB)
 endif()
 
 if(GTSAM_INSTALL_GEOGRAPHICLIB)
-  add_subdirectory(GeographicLib)
+  add_subdirectory(GeographicLib EXCLUDE_FROM_ALL)
```

Comment:
`EXCLUDE_FROM_ALL` mean that the GeographicLib will not build with "all" target. also it mean it will not build everything in GeographicLib. Only when it needed (on navigation tests) and what it needed (the lib itself).
This save from the javascript compilation error that occur otherwise.